src/tester.rs

```diff
@@ -439,15 +343,162 @@ fn write_with_colour(s: &str, colour: Color) {
     stderr.reset().ok();
 }
 
-fn output_failed() {
```

Comment:
Is the reason for removing these function calls and writing straight to `stderr` related to problems with parallelism?

Reply:
Ah, I've seen further down that you acquire a lock on `stderr`, so I'm guessing it's to ensure this happens at the correct place?

Reply:
Correct.